packages/usa-header/src/index.js

```diff
@@ -188,57 +189,83 @@ const handleEscape = (event) => {
   focusNavButton(event);
 };
 
+/**
+ * Takes clicked link and current url, removes any hash from the link, and compares them.
+ *
+ * @param {HTMLAnchorElement} targetLink - Target link clicked in header.
+ * @returns {boolean} Return `true` if target is same page.
+ */
+const isChildSection = (targetLink) => {
+  const baseURL = new URL(CURRENT_PAGE);
+  const linkDestination = new URL(targetLink.href);
+  if (baseURL.pathname === linkDestination.pathname) {
+    return true;
+  }
+
+  return false;
+};
```

Comment:
## isChildSection() logic

This test now uses the `pathname` property to compare the URL path before any hash symbols [^1]

[^1]: https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname